handler/src/main/java/io/netty/handler/ssl/ReferenceCountedOpenSslEngine.java

```diff
@@ -469,6 +469,65 @@ private int readPlaintextData(final ByteBuffer dst) {
         return sslRead;
     }
 
+    /**
+     * Visible only for testing!
+     */
+    final synchronized int maxWrapOverhead() {
+        return maxWrapOverhead;
+    }
+
+    /**
+     * Visible only for testing!
+     */
+    final synchronized int maxEncryptedPacketLength() {
+        return maxEncryptedPacketLength0();
+    }
+
+    /**
+     * This method is intentionally not synchronized, only use if you know you are in the EventLoop
+     * thread and visibility on {@link #maxWrapOverhead} is achieved via other synchronized blocks.
+     */
+    final int maxEncryptedPacketLength0() {
+        return maxWrapOverhead + MAX_PLAINTEXT_LENGTH;
+    }
+
+    /**
+     * This method is intentionally not synchronized, only use if you know you are in the EventLoop
+     * thread and visibility on {@link #maxWrapBufferSize} and {@link #maxWrapOverhead} is achieved
+     * via other synchronized blocks.
+     */
+    final int calculateMaxLengthForWrap(int plaintextLength, int numComponents) {
+        return (int) min(maxWrapBufferSize, plaintextLength + (long) maxWrapOverhead * numComponents);
```

Comment:
let me know if you think there is a problem. The intention is not to necessarily to protect this variable in the synchronization block because it is mostly accessed inside the EventLoop (except maybe for `OpenSslSession#getPacketBufferSize()`).